src/topology/order/hom.lean

```diff
@@ -0,0 +1,115 @@
+category of Priestley spaces.
+
+We use the `fun_like` design, so each type of morphisms has a companion typeclass which is meant to
+be satisfied by itself and all stricter types.
+
+## Types of morphisms
+
+* `continuous_order_hom`: Continuous monotone functions, aka Priestley homomorphisms.
+
+## Typeclasses
+
+* `continuous_order_hom_class`
+-/
+
+open function
+
+variables {F α β γ δ : Type*}
+
+/-- The type of continuous monotone maps from `α` to `β`, aka Priestley homomorphisms. -/
+structure continuous_order_hom (α β : Type*) [preorder α] [preorder β] [topological_space α]
+  [topological_space β]
+  extends order_hom α β :=
+(continuous_to_fun : continuous to_fun)
+
+infixr ` →Co `:25 := continuous_order_hom
+
+/-- `continuous_order_hom_class F α β` states that `F` is a type of continuous monotone maps.
+
+You should extend this class when you extend `continuous_order_hom`. -/
+class continuous_order_hom_class (F : Type*) (α β : out_param $ Type*) [preorder α] [preorder β]
+  [topological_space α] [topological_space β]
+  extends rel_hom_class F ((≤) : α → α → Prop) ((≤) : β → β → Prop) :=
+(map_continuous (f : F) : continuous f)
+
+--TODO@Yael: Add instance to `continuous_map_class` once we have it, add coercion to `α →Co β`
+
+/-! ### Top homomorphisms -/
+
+namespace continuous_order_hom
+variables [topological_space α] [preorder α] [topological_space β]
+
+section preorder
+variables [preorder β] [topological_space γ] [preorder γ] [topological_space δ] [preorder δ]
+
+/-- Reinterpret a `continuous_order_hom` as a `continuous_map`. -/
+def to_continuous_map (f : α →Co β) : C(α, β) := { ..f }
+
+instance : continuous_order_hom_class (α →Co β) α β :=
+{ coe := λ f, f.to_fun,
+  coe_injective' := λ f g h, by { obtain ⟨⟨_, _⟩, _⟩ := f, obtain ⟨⟨_, _⟩, _⟩ := g, congr' },
+  map_rel := λ f, f.monotone',
+  map_continuous := λ f, f.continuous_to_fun }
+
+/-- Helper instance for when there's too many metavariables to apply `fun_like.has_coe_to_fun`
+directly. -/
+instance : has_coe_to_fun (α →Co β) (λ _, α → β) := fun_like.has_coe_to_fun
+
+@[simp] lemma to_fun_eq_coe {f : α →Co β} : f.to_fun = (f : α → β) := rfl
+
+@[ext] lemma ext {f g : α →Co β} (h : ∀ a, f a = g a) : f = g := fun_like.ext f g h
+
+--TODO@Yael: Add `copy` once we have `continuous_map.copy`
+
+variables (α)
+
+/-- `id` as a `continuous_order_hom`. -/
+protected def id : continuous_order_hom α α := ⟨order_hom.id, continuous_id⟩
+
+instance : inhabited (continuous_order_hom α α) := ⟨continuous_order_hom.id _⟩
+
+@[simp] lemma coe_id : ⇑(continuous_order_hom.id α) = id := rfl
+
+variables {α}
+
+@[simp] lemma id_apply (a : α) : continuous_order_hom.id α a = a := rfl
+
+/-- Composition of `continuous_order_hom`s as a `continuous_order_hom`. -/
+def comp (f : β →Co γ) (g : α →Co β) : continuous_order_hom α γ :=
+{ ..f.to_order_hom.comp g.to_order_hom, ..f.to_continuous_map.comp g.to_continuous_map }
```

Comment:
Do we care about defeq for the `to_order_hom` projection? If yes, then you can use
```suggestion
⟨f.to_order_hom.comp g.to_order_hom, f.continuous.comp g⟩
```